File: backend/app/schemas/audit.py
"""Audit log schemas: response + filters (Section 16.1)."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class AuditLogOut(BaseModel):
    id: str
    action: str
    actor_user_id: str | None = None
    actor_email: str | None = None
    actor_name: str | None = None
    entity_type: str | None = None
    entity_id: str | None = None
    old_values: dict | None = None
    new_values: dict | None = None
    metadata: dict | None = None
    ip_address: str | None = None
    created_at: datetime


class AuditLogPage(BaseModel):
    items: list[AuditLogOut]
    total: int
    page: int
    page_size: int
    total_pages: int


class AuditActionGroup(BaseModel):
    domain: str
    actions: list[str]

File: shared/types/audit.ts
export interface AuditLog {
  id: string;
  action: string;
  actor_user_id: string | null;
  actor_email: string | null;
  actor_name: string | null;
  entity_type: string | null;
  entity_id: string | null;
  old_values: Record<string, unknown> | null;
  new_values: Record<string, unknown> | null;
  metadata: Record<string, unknown> | null;
  ip_address: string | null;
  created_at: string;
}

export interface AuditLogPage {
  items: AuditLog[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

export interface AuditActionGroup {
  domain: string;
  actions: string[];
}

export interface AuditFilter {
  search?: string;
  action?: string;
  entity_type?: string;
  page?: number;
  page_size?: number;
}
